src/Microsoft.IdentityModel.Tokens/AsymmetricAdapter.cs

```diff
@@ -296,7 +296,8 @@ private void Initialize(RSA rsa, string algorithm)
             // X509Certificate2.GetPrivateKey OR X509Certificate2.GetPublicKey.Key
             // These calls return an AsymmetricAlgorithm which doesn't have API's to do much and need to be cast.
             // RSACryptoServiceProvider is wrapped to support SHA2
-#if NET45 || NET451 || NET461 || NETSTANDARD2_0
+            // RSACryptoServiceProviderProxy is only supported on Windows platform
+#if WINDOWS && (NET45 || NET451 || NET461 || NETSTANDARD2_0)
```

Comment:
>WINDOWS [](start = 4, length = 7)

consider throwing a platform exception if !WINDOWS or alternatively do nothing (except logging that RSA is not supported).

Reply:
There is no reason to throw here.

The purpose of RSACryptoServiceProviderProxy is to _enable_ SHA2 signatures in cases when  provided RSACryptoServiceProvider understands only SHA1 - which is the case only on Windows.
On .NET Core, RSACryptoServiceProvider understands SHA2, so no proxy is needed.